gcore/gdalrasterband.cpp

```diff
@@ -1929,6 +1929,128 @@ uint64_t GDALRasterBand::GetNoDataValueAsUInt64(int *pbSuccess)
     return std::numeric_limits<uint64_t>::max();
 }
 
+/**
+ * \brief Returns a replacement value for a nodata value.
+ *
+ * @param dt Data type
+ * @param dfNoDataValue The no data value
+
+ * @since GDAL 3.10
+ */
+double GDALGetRasterNoDataReplacementValue(GDALDataType dt,
+                                           double dfNoDataValue)
+{
+
+    double dfReplacementVal = dfNoDataValue;
+    if (dt == GDT_Byte)
+    {
+        dfNoDataValue = GDALClampDoubleValue(
```

Comment:
yes, but my thinking was that the computational cost of an additional if, compared to an assignment was more or less the same, what I can do here is to make GDALClampDoubleValue modify dfNoDataValue and return TRUE if it was changed, and return 0 straight away if not.

Reply:
regarding computation cost, normally GetRasterNoDataReplacementValue shouldn't not be used inside a performance critical loop. The typical use pattern is to call it outside of the loop iterating over pixels.
Maybe the clamping works. With my mind set, it seems to be simpler to have this logic: "test if the provided nodata is in the validity range of the indicated data type. If it is not, exit early and return a 'random'=0 replace nodata value that is never going to be used anyway since no computed pixel will be equal to the nodata value since it is out of range"